src/Stack/Build/Execute.hs

```diff
@@ -1532,6 +1533,28 @@ singleBuild ac@ActionContext {..} ee@ExecuteEnv {..} task@Task {..} installedMap
                 _ -> return ()
             when hasLibrary $ cabal KeepTHLoading ["register"]
 
+        -- copy ddump-* files
+        let enableDdumpDir = isJust $ boptsDdumpDir eeBuildOpts
+            ddumpPath      = maybe "" T.unpack $ boptsDdumpDir eeBuildOpts
+        when (buildingFinals && enableDdumpDir && not (null ddumpPath)) $ do
```

Comment:
It looks like a pattern match would be easier to read here, e.g.:

```haskell
case T.unpack <$> boptsDdumpDir eebuildOpts of
  Just ddumpPath | buildingFinals && enableDdumpDir && not (null ddumpPath) -> do ...
  _ -> pure ()
```

Reply:
That makes sense. That's true :)